jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.c

```diff
@@ -103,57 +103,83 @@ ecma_builtin_string_prototype_object_value_of (ecma_value_t this_arg) /**< this
 } /* ecma_builtin_string_prototype_object_value_of */
 
 /**
- * The String.prototype object's 'charAt' routine
- *
- * See also:
- *          ECMA-262 v5, 15.5.4.4
+ * Helper function for the String.prototype object's 'charAt' and charCodeAt' routine
  *
  * @return ecma value
  *         Returned value must be freed with ecma_free_value.
  */
 static ecma_value_t
-ecma_builtin_string_prototype_object_char_at (ecma_value_t this_arg, /**< this argument */
-                                              ecma_value_t arg) /**< routine's argument */
+ecma_builtin_string_prototype_char_at_helper (ecma_value_t this_arg, /**< this argument */
+                                              ecma_value_t arg, /**< routine's argument */
+                                              bool charcode_mode) /**< routine mode */
```

Comment:
What does the true/false value mean for the charcode_mode? Maybe we need to document it, or (imho) change the type to enum values to make it more explicit.

Reply:
I'd prefer the enum too, but I wonder if that'd affect the binary size negatively?